lib/helpers.py

```diff
@@ -186,6 +186,29 @@ def uplift_report(self, marks_and_spend_df, attributions_df):
 
         return report_df
 
+    def remove_users_marked_as_control_and_test(self, marks_and_spend_df):
+        """
+        Remove users that have been added to both the test and control group.
+
+        :param marks_and_spend_df: Marks and spend dataframe
+        :type marks_and_spend_df: pandas.DataFrame
+
+        :return: Dataframe without users marked as both test and control
+        :rtype: pandas.DataFrame
+        """
+        user_to_ab_test_group_count = marks_and_spend_df[marks_and_spend_df['event_type'] == 'mark'] \
+            .groupby('user_id', as_index=False).agg({'ab_test_group': pd.Series.nunique })
+
+        users_with_double_marks = user_to_ab_test_group_count[user_to_ab_test_group_count['ab_test_group'] > 1]
+
+        number_of_users_with_double_marks = len(users_with_double_marks)
```

Comment:
maybe you can return right here if this is 0, so that we don't log `Removed 0 users`...

Reply:
I'll add a special message for no users removed. Usually a better idea to explicitly provide feedback in any case than just omitting it

Reply:
I was also thinking that maybe the filter operation on the return might be expensive, but you said you checked it with a big dataframe so it should be okay like this